### PR TITLE
Added example for doxygen comment

### PR DESCRIPTION
**Pull request by ni4**
This PR updates Developers Guide with sample doxygen comment as it was discussed in #192.
All comments, including related to English language usage, are welcome.

@flowher I saw you like lines of stars/dashes in comments, imho they are not visually appealing. However if you insist on them we can run a voting.

### CODE REVIEW

File: Developers Guide.md [hunk 1]
```diff
@@ -354,6 +354,33 @@ Do not:
...
+ *  @param [in] param2 integer, some number representing something
+ *  @param [in, out] size number of bytes available to store in buffer
+ *  @param [out] buffer buffer to store results, may be NULL. In this case size can be used to 
+ *                      obtain the required buffer length
+ *  @return 0 if operation succeeds, or error code otherwise. If operation succeeds then buffer 
+ *          is populated with the resulting data, and size contains the length of this data.
+ *          if error code is E_BUF_TOOSMALL then size will contain the required size to store
+ *          the result
+ **/
+uint32_t
```
Comment (flowher) [inline Developers Guide.md]: type for error codes is called rnp_error

File: Developers Guide.md [hunk 2]
```diff
@@ -354,6 +354,33 @@ Do not:
...
 * Do not use `pragma`, and try to avoid `__attribute__` as well.
 
+### Documentation
+Documentation is done in Doxygen comments format, which must be put in header files.
+Exception are static or not exposed to the public API functions - they should be documented in 
+the code.
+
+Comments should use doxygen markdown style, like the following example:
+
+```c
```
Comment (flowher) [inline Developers Guide.md]: 'c' like comment :) maybe not needed?

**Comment by dewyatt** [comment]
Great! Some minor suggestions:

1. I think we may as well drop the `@brief` markup since we're using javadoc-style comments, we can just turn on `JAVADOC_AUTOBRIEF` and use:

```c
/** your brief description here
 *
 *  Details here
[...]
 */
```

2. > Exception are static or not exposed to the public API functions - they should be documented in the code.

Can you also add a note to add an `@internal` tag for these?

3. We should add a minimal per-file header as well, something similar to this:

```c
/** PGP packet parsing utilities
 *  @file
 */
```

File: Developers Guide.md [hunk 3]
```diff
@@ -354,6 +354,33 @@ Do not:
   ```
 * Do not use `pragma`, and try to avoid `__attribute__` as well.
 
+### Documentation
+Documentation is done in Doxygen comments format, which must be put in header files.
+Exception are static or not exposed to the public API functions - they should be documented in
```
Comment (flowher) [inline Developers Guide.md]: As per our voting result, please remove "or not exposed to the public API functions". You can use instead "not having declaration"  or "having only definition" (generally I mean those which are not in .h for some reason).

File: Developers Guide.md (hunk 1 above)
Comment (ni4) [inline Developers Guide.md]: ok

File: Developers Guide.md (hunk 2 above)
Comment (ni4) [inline Developers Guide.md]: It's just to have a nice coloring of the function definition below :)

Comment (flowher) [inline Developers Guide.md]: ah! wow didn't know this feature. cool

File: Developers Guide.md (hunk 3 above)
Comment (ni4) [inline Developers Guide.md]: Ok.

**Comment by ni4** [comment]
@dewyatt, @flowher please see the updated documentation, it seems I didn't miss any of your suggestions. GitHub PR reviews/comments sometimes hard to follow.

**Comment by dewyatt** [comment]
Any thoughts on dropping the param direction (`[in]/[out]`) piece?
It kind of seems redundant in C.
```c
rnp_result some_func(int abc, // [in]
                     int *def, // [out]
                     const char *s);   // [in]
```
I guess the only time it isn't clear is when you have `[in,out]`, which seems pretty rare.

Anyways, LGTM. This is sort of a policy update I guess so let's get @ronaldtse to approve.

**Comment by ronaldtse** [comment]
`[in] / [out]` might not be necessary, if the param description already describes what that does. In the case of `[in, out]`, it would actually be much cleaner for a function not to have an in & out argument, but simply in or out.

**Comment by ni4** [comment]
@ronaldtse  @dewyatt  Agreed on [in, out]. Removed them, so if there are no any other suggestions let's merge this.

**Comment by dewyatt** [comment]
> it would actually be much cleaner for a function not to have an argument used both for in & out. Maybe we should specify this when we drop the requirement for [in] / [out].

Definitely agree, botan's FFI interface has some in/out parameters and it has puzzled me before (partially a documentation issue there though).
